iosxr_iso2vbox.py

```diff
@@ -69,29 +69,54 @@
 import argparse
 from argparse import RawDescriptionHelpFormatter
 import re
+import logging
 
 # Telnet ports used to access IOS XR via socat
 console_port = 65000
 aux_port = 65001
 
 
-def run(command_string, debug=False):
+def run_process_quietly(cmd_string, debug=False):
     '''
-    Execute a CLI command string and return the result.
+    Execute a CLI command in the 'background' as process.
+    '''
+    argv = shlex.split(cmd_string)
+    if debug is True:
+        print ('argv: %s' % argv)
+    with open(os.devnull, 'w') as fp:
+        subprocess.Popen((argv), stdout=fp)
+    time.sleep(2)
 
-    Note this does not handle operators which need whitespace.
 
+def run_get_output(cmd_string, debug=False):
+    '''
+    Execute a CLI command string and return the output.
+    Note this does not handle operators which need whitespace.
     In those cases the raw commands will need to be used.
-
     E.g. passing grep 'a string separated by spaces'.
     '''
-    argv = command_string.split()
+    argv = cmd_string.split()
     if debug is True:
         print ('argv: %s' % argv)
     result = subprocess.check_output(argv)
     return result
 
 
+def run(cmd_string, debug=False):
+    '''
+    Execute a CLI command string quietly and return the result.
+    Note this does not handle operators which need whitespace.
+    In those cases the raw commands will need to be used.
+    E.g. passing grep 'a string separated by spaces'.
+    This is the bash equivalent of command > /dev/null.
+    '''
+    argv = cmd_string.split()
+    if debug is True:
+        print ('argv: %s' % argv)
+    FNULL = open(os.devnull, 'w')
```

Comment:
That's a bit like saying this code is bugged and I know that 8)

On 7/14/16, 3:36 PM, Richard Wellum wrote:
>>       if debug is True:
>>           print ('argv: %s' % argv)
>>       result = subprocess.check_output(argv)
>>       return result
>>
>>
>> +def run(cmd_string, debug=False):
>> +    '''
>> +    Execute a CLI command string quietly and return the result.
>> +    Note this does not handle operators which need whitespace.
>> +    In those cases the raw commands will need to be used.
>> +    E.g. passing grep 'a string separated by spaces'.
>> +    This is the bash equivalent of command > /dev/null.
>> +    '''
>> +    argv = cmd_string.split()
>> +    if debug is True:
>> +        print ('argv: %s' % argv)
>> +    FNULL = open(os.devnull, 'w')
>
> I call that out:
>
> Execute a CLI command string quietly and return the result.
> Note this does not handle operators which need whitespace.
>
> ---
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/ios-xr/iosxrv-x64-vbox/pull/10/files/fc29a814958e64ddff6a567916527f1c4084fe76#r70869150
>